# 📊 Análise Descritiva de Disparos Policiais Fatais nos EUA (2015-2020) 🚔

## Projeto Final | Turma 1015 | Programa BeTech by Braskem
### Alunos: Caio Henrique e Daniel Costa

Bem-vindos ao nosso projeto final de análise descritiva! Aqui, exploraremos os dados dos disparos policiais fatais ocorridos nos Estados Unidos entre 2015 e 2020. Vamos mergulhar nessa análise para entender melhor o cenário desses eventos.

## 🎯 Objetivo do Projeto

Nosso objetivo é examinar os dados disponíveis e obter insights sobre os disparos policiais fatais nos EUA. Usando técnicas de análise descritiva, investigaremos os padrões, identificaremos as características desses incidentes e entenderemos melhor os fatores que os cercam. Queremos contribuir para uma análise mais abrangente e promover discussões relevantes sobre a questão da violência policial. 📈👮‍♂️

## ℹ️ Sobre os Dados

Os dados que utilizaremos são abrangentes e percorrem o período de 2015 a 2020. Vamos explorar informações como idade, raça, gênero e outras variáveis relevantes. Esses dados nos ajudarão a entender melhor a dinâmica dos disparos policiais fatais nos EUA. 📅💻

Os dados foram obtidos a partir do conjunto de dados "Fatal Police Shootings" disponível no Kaggle. Você pode acessar o dataset [aqui](https://www.kaggle.com/datasets/andrewmvd/police-deadly-force-usage-us?select=fatal-police-shootings-data.csv).


## 📈 Análise Descritiva

Nossa análise descritiva envolverá a exploração dos seguintes aspectos:

1. **Tipos de Colunas:** Vamos identificar os diferentes tipos de dados presentes em nosso conjunto de dados, incluindo variáveis numéricas, categóricas e textuais.

2. **Valores Nulos:** Investigaremos a quantidade de valores nulos em cada coluna, a fim de avaliar a qualidade dos dados e determinar quais tratamentos podem ser necessários.

3. **Valores Únicos em Colunas Categóricas:** Analisaremos o número de valores únicos presentes em colunas categóricas, fornecendo insights sobre a diversidade dessas variáveis.

## 🌍 Importância e Impacto

O tema dos disparos policiais fatais nos EUA é de extrema relevância e impacto social. Por meio de uma análise descritiva aprofundada, buscamos contribuir para uma compreensão mais ampla desses eventos, estimulando o diálogo construtivo e a busca por soluções que promovam a justiça e a segurança para todos. 🌟✊

Vamos embarcar nesta jornada de análise descritiva para desvendar os insights ocultos nesses dados? Venha conosco e vamos explorar juntos esse importante problema social! 📊🔍💡

*Aviso: Os dados utilizados neste projeto são baseados em eventos reais e podem conter informações sensíveis e perturbadoras. Trataremos esses dados com a devida seriedade e respeito.*


# Análise Descritiva📊🔍

Nesta seção, realizaremos uma análise descritiva do conjunto de dados para obter insights sobre os disparos policiais fatais nos EUA.

**Tipos de Colunas:** ✍️📋

O conjunto de dados possui 14 colunas com os seguintes tipos:


- **id**: integer
- **name** (nome): text
- **date_of_death** (data de falecimento): date
- **manner_of_death** (modo de morte): text
- **armed** (armado): text
- **age** (idade): integer
- **gender** (gênero): char
- **race** (raça): char
- **city** (cidade): text
- **state** (estado): text
- **signs_of_mental_illness** (sinais de doença mental): boolean
- **threat_level** (nível de ameaça): text
- **flee** (fuga): text
- **body_camera** (câmera corporal): boolean

Essa descrição fornece uma visão geral dos tipos de dados em cada coluna do conjunto de dados e será útil para realizar nossa análise posteriormente.

**Valores Nulos:** ❓🔎

Após a realização da consulta, identificamos os seguintes valores nulos em cada coluna:

- **armed**: 240 valores nulos
- **age**: 249 valores nulos
- **gender**: 2 valores nulos
- **race**: 607 valores nulos
- **flee**: 250 valores nulos


**Valores Únicos em Colunas Categóricas:** 🧮🔍

- **id:** 5.338 valores únicos
- **nome:** 5.115 valores únicos
- **date_of_death:** 1.823 valores únicos
- **manner_of_death:** 2 valores únicos
- **age:** 77 valores únicos
- **gender:** 2 valores únicos
- **city:** 2.447 valores únicos
- **state:** 51 valores únicos
- **signs_of_mental_illness:** 2 valores únicos
- **threat_level:** 3 valores únicos
- **flee:** 4 valores únicos
- **body_camera:** 2 valores únicos

Esses números evidenciam a diversidade de informações presentes nas colunas e fornecem insights valiosos sobre a amplitude e complexidade dos dados analisados.

In [85]:
# Checando a quantidade de valores nulos e valores únicos em colunas categóricas

from sqlalchemy import create_engine
import pandas as pd

engine = create_engine('postgresql+psycopg2://postgres:123@localhost/postgres')
conn = engine.connect()

query = """
SELECT * FROM projeto.fatal_shootings;
"""
df = pd.read_sql_query(query, con = conn)
null_values = df.isnull().sum()
print(f"número de valores nulos: \n{null_values}\n")


id = df['id'].nunique()
name = df['name'].nunique()
date_of_death = df['date_of_death'].nunique()
manner_of_death = df['manner_of_death'].nunique()
armed = df['armed'].nunique()
age = df['age'].nunique()
gender = df['gender'].nunique()
city = df['city'].nunique()
state = df['state'].nunique()
signs_of_mental_illness = df['signs_of_mental_illness'].nunique()
threat_level = df['threat_level'].nunique()
flee = df['flee'].nunique()
body_camera = df['body_camera'].nunique()

print(f"id = {id}")
print(f"nome = {name}")
print(f"date_of_death = {date_of_death}")
print(f"manner_of_death = {manner_of_death}")
print(f"age = {age}")
print(f"gender = {gender}")
print(f"city = {city}")
print(f"state = {state}")
print(f"signs_of_mental_illness = {signs_of_mental_illness}")
print(f"threat_level = {threat_level}")
print(f"flee = {flee}")
print(f"body_camera = {body_camera}")



número de valores nulos: 
id                           0
name                         0
date_of_death                0
manner_of_death              0
armed                      240
age                        249
gender                       2
race                       607
city                         0
state                        0
signs_of_mental_illness      0
threat_level                 0
flee                       250
body_camera                  0
dtype: int64

id = 5338
nome = 5115
date_of_death = 1823
manner_of_death = 2
age = 77
gender = 2
city = 2447
state = 51
signs_of_mental_illness = 2
threat_level = 3
flee = 4
body_camera = 2


# Consultas📊🔍

Nesta seção, responderemos a algumas consultas relevantes sobre o conjunto de dados dos disparos policiais fatais nos EUA.

## Consulta 1: Distribuição Demográfica

Vamos analisar a distribuição demográfica das vítimas, incluindo idade, gênero e raça.

### Perguntas:
- Qual é a faixa etária mais comum das vítimas?
- Quais são os gêneros mais representados nas ocorrências?
- Como é a distribuição racial das vítimas?


In [ ]:
# Qual é a faixa etária mais comum das vítimas?

from sqlalchemy import create_engine
import pandas as pd

engine = create_engine('postgresql+psycopg2://postgres:123@localhost/postgres')
conn = engine.connect()

query = """
SELECT age AS idade, count(id) AS total
FROM projeto.fatal_shootings
GROUP BY age
ORDER BY total DESC
LIMIT 10;
"""
df = pd.read_sql_query(query, con = conn)
df

idade  total
0    NaN    249
1   25.0    180
2   31.0    178
3   33.0    169
4   29.0    169
5   27.0    169
6   34.0    166
7   32.0    166
8   24.0    154
9   36.0    153

É possível perceber que apesar de 249 registros não terem idade confirmada, a faixa etária mais comum das vítimas de disparos policiais fatais nos EUA varia de 25 a 35 anos. Essas informações nos ajudam a entender melhor o perfil das vítimas nesse contexto.

In [ ]:
# Quais são os gêneros mais representados nas ocorrências?

from sqlalchemy import create_engine
import pandas as pd

engine = create_engine('postgresql+psycopg2://postgres:123@localhost/postgres')
conn = engine.connect()

query = """
SELECT gender AS gênero, count(id) AS total
FROM projeto.fatal_shootings
GROUP BY gender
ORDER BY total DESC;
"""
df = pd.read_sql_query(query, con = conn)
df

gênero  total
0      M   5102
1      F    234
2   None      2

A maioria esmagadora dos casos se dão com pessoas do sexo masculino, com uma fatia de mais de 90% dos casos.

In [ ]:
# Como é a distribuição racial das vítimas?

from sqlalchemy import create_engine
import pandas as pd

engine = create_engine('postgresql+psycopg2://postgres:123@localhost/postgres')
conn = engine.connect()

query = """
SELECT race AS raça, COUNT(*) AS total
FROM (
    SELECT CASE WHEN (race IS null OR race LIKE 'O') THEN 'desconhecida' ELSE race END AS race
    FROM projeto.fatal_shootings
    ) AS subquery
GROUP BY race
ORDER BY total DESC;
"""
df = pd.read_sql_query(query, con = conn)
df

raça  total
0             W   2385
1             B   1254
2             H    878
3  desconhecida    654
4             A     91
5             N     76

A raça mais comum entre as vítimas é a raça branca (W), com um total de 2385 ocorrências seguida pela a raça negra (B) com 1254 ocorrências e a raça hispânica (H) com 878 ocorrências. A categoria "desconhecida" agrupa as raças "O" (Outro) e nulos, e apresenta um total de 654 ocorrências.
Esses resultados fornecem uma visão inicial sobre a distribuição das raças entre as vítimas. É importante ressaltar que essa análise é baseada nos dados disponíveis no conjunto de dados analisado e não reflete necessariamente a representação demográfica completa da população.

É possível explorar esta possível relação a partir da comparação entre datasets que detalhem a composição racial e a densidade demográfica nas cidades e estados presentes no conjunto de dados.

## Consulta 2: Armas Utilizadas

Vamos explorar as armas utilizadas nas ocorrências.

### Perguntas:
- Quais são as armas mais comuns envolvidas nas ocorrências?
- Qual é a proporção de suspeitos armados e desarmados nas ocorrências?

In [ ]:
# Quais são as armas mais comuns envolvidas nas ocorrências?

from sqlalchemy import create_engine
import pandas as pd

engine = create_engine('postgresql+psycopg2://postgres:123@localhost/postgres')
conn = engine.connect()

query = """
SELECT armed AS arma, COUNT(armed) AS total
FROM projeto.fatal_shootings
GROUP BY armed
ORDER BY total DESC
LIMIT 10;
"""
df = pd.read_sql_query(query, con = conn)
df

arma  total
0             gun   2997
1           knife    780
2         unarmed    318
3    undetermined    221
4      toy weapon    177
5         vehicle    125
6  unknown weapon     74
7         machete     44
8           Taser     26
9              ax     24

As armas mais comuns envolvidas nas ocorrências de disparos policiais fatais são as armas de fogo, presentes em 2997 casos. Em seguida, as facas, presentes em 780 casos e armas de brinquedo estão presentes em 177 casos.
Vale mencionar que, apesar dessas serem as armas mais frequentes, também ocorreram situações em que os suspeitos não estavam armados (318 casos) e em que a presença de arma foi classificada como indeterminada (221 casos).

In [ ]:
# Qual é a proporção de suspeitos armados e desarmados nas ocorrências?

from sqlalchemy import create_engine
import pandas as pd

engine = create_engine('postgresql+psycopg2://postgres:123@localhost/postgres')
conn = engine.connect()

query = """
SELECT armed AS armado, COUNT(armed) AS total
FROM (
  SELECT CASE
   WHEN armed != 'unarmed' AND armed != 'undetermined'
  THEN 'armed'
  ELSE armed END AS armed
  FROM projeto.fatal_shootings
  ) AS subquery
GROUP BY armed
ORDER BY total DESC;
"""
df = pd.read_sql_query(query, con = conn)
df

armado  total
0         armed   4559
1       unarmed    318
2  undetermined    221
3          None      0

A maioria dos casos envolve suspeitos armados com 4559 ocorrências, seguidos por 318 casos de suspeitos desarmados e 221 casos em que o status é classificado como indeterminado.

Essas informações ressaltam a presença significativa de suspeitos armados nos casos analisados. No entanto, também é relevante notar que houve ocorrências em que os suspeitos estavam desarmados ou em que a classificação do status foi inconclusiva.

## Consulta 3: Circunstâncias dos Incidentes
Vamos investigar as circunstâncias dos incidentes.

## Perguntas:
- Quais são os níveis de ameaça mais frequentes relatados nos incidentes?
- Como é a distribuição dos casos em relação ao uso de câmera corporal por parte dos policiais?
- Houve relato de sinais de doença mental nos casos?
- Qual é a incidência de tiroteios fatais envolvendo suspeitos que estavam em fuga?

In [ ]:
# Quais são os níveis de ameaça mais frequentes relatados nos incidentes?

from sqlalchemy import create_engine
import pandas as pd

engine = create_engine('postgresql+psycopg2://postgres:123@localhost/postgres')
conn = engine.connect()

query = """
SELECT threat_level AS nível_ameaça, COUNT(*) AS total
FROM projeto.fatal_shootings
GROUP BY threat_level
ORDER BY total desc
LIMIT 5;
"""
df = pd.read_sql_query(query, con = conn)
df

nível_ameaça  total
0        attack   3434
1         other   1643
2  undetermined    261

A maioria dos casos, com 3434 ocorrências, indica uma situação de perigo significativo. Além disso, houve um número considerável de casos classificados como "outro", cuja natureza exata não está especificada nos dados disponíveis. Por fim, houveram 261 casos em que a natureza do incidente foi classificada como indeterminada, o que sugere a necessidade de uma análise mais aprofundada para entender melhor esses casos.

In [3]:
# Como é a distribuição dos casos em relação ao uso de câmera corporal por parte dos policiais?

from sqlalchemy import create_engine
import pandas as pd

engine = create_engine('postgresql+psycopg2://postgres:123@localhost/postgres')
conn = engine.connect()

query = """
SELECT body_camera AS câmera_corporal, COUNT(*) AS total
FROM projeto.fatal_shootings
GROUP BY body_camera
ORDER BY total desc
LIMIT 5;
"""
df = pd.read_sql_query(query, con = conn)
df

câmera_corporal  total
0            False   4739
1             True    599

Dos casos analisados, em 4739 deles a câmera corporal não estava sendo utilizada durante o incidente. Em contraste, em 599 casos, a presença de câmera corporal foi registrada. Essa diferença pode ter implicações na disponibilidade de evidências e na transparência dos eventos. Além disso, observa-se uma tendência de maior número de casos com vítimas quando a câmera não é utilizada, sugerindo a importância do uso de câmeras corporais para documentar e esclarecer os eventos durante as ocorrências de disparos policiais fatais.

In [10]:
# Houve relato de sinais de doença mental nos casos?

from sqlalchemy import create_engine
import pandas as pd

engine = create_engine('postgresql+psycopg2://postgres:123@localhost/postgres')
conn = engine.connect()

query = """
SELECT signs_of_mental_illness AS doenca_mental, COUNT(*) AS total
FROM projeto.fatal_shootings
GROUP BY signs_of_mental_illness
ORDER BY total desc
LIMIT 5;
"""
df = pd.read_sql_query(query, con = conn)
df

doenca_mental  total
0          False   4143
1           True   1195

Dos casos de tiroteios fatais, 1.195 relataram sinais de doença mental, enquanto 4.143 não tiveram relatos desses sinais.

In [88]:
# Qual é a incidência de tiroteios fatais envolvendo suspeitos que estavam em fuga da polícia em comparação com aqueles que não estavam em fuga?

from sqlalchemy import create_engine
import pandas as pd

engine = create_engine('postgresql+psycopg2://postgres:123@localhost/postgres')
conn = engine.connect()

query = """
SELECT flee AS fuga, COUNT(*) AS total
FROM projeto.fatal_shootings
GROUP BY flee
ORDER BY total desc
LIMIT 5;
"""
df = pd.read_sql_query(query, con = conn)
df

fuga  total
0  Not fleeing   3356
1          Car    889
2         Foot    684
3         None    250
4        Other    159

A maioria esmagadora dos tiroteios fatais ocorreu quando os suspeitos não estavam em fuga da polícia, estes foram 3356 casos. Outros 889 casos envolveram suspeitos fugindo de carro e a pé, com 684 casos.

## Consulta 4: Cidades e Estados
Nesta consulta, analisaremos a localização dos tiroteios fatais, focalizando cidades e estados envolvidos.

### Perguntas:
1. Quais cidades têm o maior número de tiroteios fatais?
2. Quais estados têm a maior incidência de tiroteios fatais?
3. Qual foi a data com o maior número de tiroteios fatais registrados no dataset?

In [58]:
# Quais cidades têm o maior número de tiroteios fatais?

from sqlalchemy import create_engine
import pandas as pd

engine = create_engine('postgresql+psycopg2://postgres:123@localhost/postgres')
conn = engine.connect()

query = """
SELECT city AS cidade, COUNT(city) AS total
FROM projeto.fatal_shootings
GROUP BY city
ORDER BY total desc
LIMIT 5;
"""
df = pd.read_sql_query(query, con = conn)
df

cidade  total
0  Los Angeles     83
1      Phoenix     75
2      Houston     56
3    Las Vegas     44
4  San Antonio     43

Os resultados mostram as cinco cidades dos EUA com maior número de tiroteios fatais entre 2015 e 2020: Los Angeles (83), Phoenix (75), Houston (56), Las Vegas (44) e San Antonio (43)

In [60]:
# Quais estados têm a maior incidência de tiroteios fatais?

from sqlalchemy import create_engine
import pandas as pd

engine = create_engine('postgresql+psycopg2://postgres:123@localhost/postgres')
conn = engine.connect()

query = """
SELECT state AS estado, COUNT(state) AS total
FROM projeto.fatal_shootings
GROUP BY state
ORDER BY total desc
LIMIT 5;
"""
df = pd.read_sql_query(query, con = conn)
df

estado  total
0     CA    784
1     TX    476
2     FL    344
3     AZ    251
4     CO    191

Os estados dos EUA com maior incidência de tiroteios fatais entre 2015 e 2020 são: Califórnia (784) e Texas (476). Ambos na região sudoeste do país.

In [72]:
# Qual foi a data com o maior número de tiroteios fatais registrados no dataset?

from sqlalchemy import create_engine
import pandas as pd

engine = create_engine('postgresql+psycopg2://postgres:123@localhost/postgres')
conn = engine.connect()

query = """
SELECT date_of_death AS data, COUNT(*) AS total
FROM projeto.fatal_shootings
GROUP BY date_of_death
ORDER BY total DESC
LIMIT 10;
"""
df = pd.read_sql_query(query, con = conn)
df

data  total
0  2018-06-29      9
1  2019-01-28      9
2  2018-02-01      9
3  2018-04-01      9
4  2018-01-06      9
5  2019-12-10      8
6  2017-02-10      8
7  2015-12-14      8
8  2018-04-05      8
9  2017-01-24      8

De acordo com o relatório, os dias 29 de junho de 2018, 28 de janeiro de 2019, 1º de fevereiro de 2018, 1º de abril de 2018 e 6 de janeiro de 2018 foram os que apresentaram o maior número de tiroteios fatais registrados, totalizando 9 ocorrências cada.